imap_processing/lo/l0/compression_tables.py

```diff
@@ -0,0 +1,853 @@
+    61: 61,
+    62: 62,
+    63: 63,
+    64: 64,
+    65: 65,
+    66: 66,
+    67: 67,
+    68: 68,
+    69: 69,
+    70: 70,
+    71: 71,
+    72: 72,
+    73: 73,
+    74: 74,
+    75: 75,
+    76: 76,
+    77: 77,
+    78: 78,
+    79: 79,
+    80: 80,
+    81: 81,
+    82: 82,
+    83: 83,
+    84: 84,
+    85: 85,
+    86: 86,
+    87: 87,
+    88: 88,
+    89: 89,
+    90: 90,
+    91: 91,
+    92: 92,
+    93: 93,
+    94: 94,
+    95: 95,
+    96: 96,
+    97: 97,
+    98: 98,
+    99: 99,
+    100: 100,
+    101: 104,
+    102: 109,
+    103: 114,
+    104: 118,
+    105: 123,
+    106: 129,
+    107: 134,
+    108: 140,
+    109: 146,
+    110: 152,
+    111: 159,
+    112: 165,
+    113: 172,
+    114: 180,
+    115: 187,
+    116: 195,
+    117: 204,
+    118: 212,
+    119: 221,
+    120: 231,
+    121: 241,
+    122: 251,
+    123: 262,
+    124: 273,
+    125: 284,
+    126: 296,
+    127: 309,
+    128: 322,
+    129: 336,
+    130: 350,
+    131: 365,
+    132: 381,
+    133: 397,
+    134: 414,
+    135: 432,
+    136: 450,
+    137: 469,
+    138: 489,
+    139: 510,
+    140: 532,
+    141: 555,
+    142: 579,
+    143: 603,
+    144: 629,
+    145: 656,
+    146: 684,
+    147: 713,
+    148: 743,
+    149: 775,
+    150: 808,
+    151: 843,
+    152: 879,
+    153: 916,
+    154: 956,
+    155: 996,
+    156: 1039,
+    157: 1083,
+    158: 1129,
+    159: 1178,
+    160: 1228,
+    161: 1280,
+    162: 1335,
+    163: 1392,
+    164: 1452,
+    165: 1514,
+    166: 1578,
+    167: 1646,
+    168: 1716,
+    169: 1789,
+    170: 1866,
+    171: 1945,
+    172: 2029,
+    173: 2115,
+    174: 2206,
+    175: 2300,
+    176: 2398,
+    177: 2500,
+    178: 2607,
+    179: 2719,
+    180: 2835,
+    181: 2956,
+    182: 3082,
+    183: 3214,
+    184: 3351,
+    185: 3494,
+    186: 3644,
+    187: 3799,
+    188: 3962,
+    189: 4131,
+    190: 4307,
+    191: 4491,
+    192: 4683,
+    193: 4883,
+    194: 5092,
+    195: 5310,
+    196: 5536,
+    197: 5773,
+    198: 6020,
+    199: 6277,
+    200: 6545,
+    201: 6824,
+    202: 7116,
+    203: 7420,
+    204: 7737,
+    205: 8068,
+    206: 8412,
+    207: 8772,
+    208: 9147,
+    209: 9537,
+    210: 9945,
+    211: 10370,
+    212: 10813,
+    213: 11275,
+    214: 11756,
+    215: 12259,
+    216: 12783,
+    217: 13329,
+    218: 13898,
+    219: 14492,
+    220: 15111,
+    221: 15757,
+    222: 16430,
+    223: 17132,
+    224: 17864,
+    225: 18627,
+    226: 19423,
+    227: 20253,
+    228: 21118,
+    229: 22021,
+    230: 22962,
+    231: 23943,
+    232: 24966,
+    233: 26032,
+    234: 27145,
+    235: 28304,
+    236: 29514,
+    237: 30775,
+    238: 32090,
+    239: 33461,
+    240: 34890,
+    241: 36381,
+    242: 37936,
+    243: 39556,
+    244: 41247,
+    245: 43009,
+    246: 44846,
+    247: 46763,
+    248: 48761,
+    249: 50844,
+    250: 53016,
+    251: 55282,
+    252: 57644,
+    253: 60107,
+    254: 62675,
+    255: 65353,
+}
```

Comment:
Is there any way to generate these numbers algorithmically? Instead of using these tables? Alternatively, could you write a function that has some kind of `return bit12_to_bit16[number] if number in bit12_to_bit16 else number` check instead of mapping one number to itself so many times?

Reply:
I can see if Lo can provide an algorithm for how those mappings were generated. I think I'm confused about what you mean by "instead of mapping one number to itself so many times".

As a side note, I think I should remove these  dictionaries from this PR either way because these are for the Science Count decompression. I originally started this PR thinking it would only be adding decompression tables but decided to make it a Science DE data class / decompression PR and figured I'd leave these in here, but it'd probably be better to remove them and add them in a separate Science Count data class / decompression PR. I do want to understand what you're thinking for the future though.

Reply:
for the first 100 values, you're just mapping `n:n` for example `1:1`, `2:2`... so instead of writing them out in this dictionary, if you abstract it using some kind of `convert_bits` method, you can say `if n <= 100: return n else: return bit12_to_bit16[n]` which is more clear and reduces the lines of code here. It does _feel_ like there's some pattern here though, maybe Lo can tell you more and these dictionaries will be unneeded.

Reply:
Ah, I see what you're saying. Yeah, I'll do that